### ISSUE TITLE
Getting inferior results compared to Lightgbm

### ISSUE THREAD

**Issue by munitech4u** [catboost/catboost#383]
Problem: {Not able to match to lightgbm results in Catboost:

I am trying to replicate lightgbm results from this Kaggle kernel: 
[https://www.kaggle.com/laurenstc/top-2-of-leaderboard-advanced-fe]
They are using lightgbm settings:
```
lgbm_model = LGBMRegressor(objective='regression',num_leaves=5,
                              learning_rate=0.05, n_estimators=720,
                              max_bin = 55, bagging_fraction = 0.8,
                              bagging_freq = 5, feature_fraction = 0.2319,
                              feature_fraction_seed=9, bagging_seed=9,
                              min_data_in_leaf =6, min_sum_hessian_in_leaf = 11,depth=6)
cv_rmse(lgbm_model).mean()
```

This gave me: 0.10631520515085766

I tried Catboost settings:

```
cat_model = CatBoostRegressor(learning_rate=0.05, n_estimators=720,loss_function='RMSE')
cv_rmse(cat_model).mean()
```

This gave me: 0.153500644136011

What is it, that I am not doing right?
})
catboost version: {0.8.1.1}
Operating System: {Windows 10}
CPU: {i7}
# GPU: {NA}

**Comment by annaveronika** [comment]
This might be because of overfitting or underfitting. Could you provide graph of how loss function on eval dataset changes during training? This might be because you need different learning rate - learning rate 0.05 + number of iterations 720 works for LightGBM, you need to look on how error changes and select enough iterations overcome underfitting. If you want the same number of iterations you need to increase learning rate if you observe underfitting.

**Comment by annaveronika** [comment]
You can use eval_metrics method (https://tech.yandex.com/catboost/doc/dg/concepts/python-reference_catboost_eval-metrics-docpage/) with plot=True to see the graphs.

**Comment by munitech4u** [comment]
Ok. I couldn't get eval_metrics to work. But attaching results from cv:

```
pool = Pool(X, y)
scores = cv(pool,params={'iterations': 1000, 'depth': 4, 'loss_function': 'RMSE', 'logging_level': 'Silent'})
```
Results:

![train_vs_vald](https://user-images.githubusercontent.com/7777086/41409511-186a87f2-6ff4-11e8-8f21-06fa3f4bf671.jpg)

The minimum rmse, I am able to achieve in train - 0.1258866
and vald is - 0.1649336

**Comment by annaveronika** [comment]
The training has not jet converged - you need to train until you see overfitting, when test error will start to grow. So you can increase number of iterations or increase learning rate. The rule is the following the less is the learning rate the better is the quality, but the more iterations you need for convergence.

What was the problem with eval_metrics? Here is the instruction to enable graph plotting in jupyter notebook: https://tech.yandex.com/catboost/doc/dg/installation/python-installation-additional-data-visualization-packages-docpage/
After you run this, visualization should show up. Did it not work for you?
By the way cv also has parameter plot=True, so it can also show you the graphs.

**Comment by sergmiller** [comment]
Hi @munitech4u

As I understand you tried to set catboost parameters similar with lightgbm.

In order to decrease bias and training overfit (because you have only 5K samples) you should try to change parameters bagging_temperature, rsm, l2_leaf_reg or decrease number of leaf_estimation_iterations and depth.

The key point which @annaveronika mentioned that training has not yet converged so you should set significantly bigger number of iterations like 100K or more.

I used this parameters:

`catboost.CatBoostRegressor(learning_rate=5e-3,n_estimators=200000, depth=2, leaf_estimation_iterations=1, rsm=0.5, bagging_temperature=4, loss_function='RMSE', random_state=0)`

and achieved train rmse 0.038, valid rmse 0.1071  (the same as lgbm and xgb, but  I m sure it can be better)
